articles/azure-monitor/platform/data-collector-api.md

```diff
@@ -548,6 +548,94 @@ post_data(customer_id, shared_key, body, log_type)
+import org.apache.http.impl.client.HttpClients;
+import org.springframework.http.MediaType;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.text.SimpleDateFormat;
+import java.util.Base64;
+import java.util.Calendar;
+import java.util.TimeZone;
+
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+
+public class ApiExample {
+
+  private static final String customerUUID = "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx";
+  private static final String sharedKey = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx";
+  private static final String logName = "DemoExample";
+  private static final String timestamp = "";
+  private static final String json = "{\"name\": \"test\",\n" + "  \"id\": 1\n" + "}";
+  private static final String RFC_1123_DATE = "EEE, dd MMM yyyy HH:mm:ss z";
+
+  public static void main(String[] args) throws IOException, NoSuchAlgorithmException, InvalidKeyException {
+    String dateString = getServerTime();
+    String httpMethod = "POST";
+    String contentType = "application/json";
+    String xmsDate = "x-ms-date";
+    String resource = "/api/logs";
+    String stringToHash =
+        httpMethod + "\n" + json.getBytes(StandardCharsets.UTF_8).length + "\n" + contentType + "\n" + xmsDate + ":"
+            + dateString + "\n" + resource;
+    String hashedString = getHMAC254(stringToHash, sharedKey);
+    String signature = "SharedKey " + customerUUID + ":" + hashedString;
+
+    postData(signature, dateString, json);
+  }
+
+  private static String getServerTime() {
+    Calendar calendar = Calendar.getInstance();
+    SimpleDateFormat dateFormat = new SimpleDateFormat(RFC_1123_DATE);
+    dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
+    return dateFormat.format(calendar.getTime());
+  }
+
+  private static void postData(String signature, String dateString, String json) throws IOException {
+    String url = "https://" + customerUUID + ".ods.opinsights.azure.com/api/logs?api-version=2016-04-01";
+    HttpPost httpPost = new HttpPost(url);
+    httpPost.setHeader("Authorization", signature);
+    httpPost.setHeader(CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+    httpPost.setHeader("Log-Type", logName);
+    httpPost.setHeader("x-ms-date", dateString);
+    httpPost.setHeader("time-generated-field", timestamp);
+    httpPost.setEntity(new StringEntity(json));
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+    HttpResponse response = httpClient.execute(httpPost);
+
+    int statusCode = response.getStatusLine().getStatusCode();
+    System.out.println(statusCode);
+  }
+
+  private static String getHMAC254(String input, String key) throws InvalidKeyException, NoSuchAlgorithmException {
+    Mac sha254HMAC;
+    String hash;
+    sha254HMAC = Mac.getInstance("HmacSHA256");
```

Comment:
Nit: do not leave uninitialized variables laying around. Lines 623 and 624 can be combined with 625 and 630 respectively. Since there's no code path that may need to respond to an uninitialized state of these variables, there's no need to have them.